client/client-multi/client-c/src/platform-impl/kaa_tcp_channel.c

```diff
@@ -186,6 +194,15 @@ kaa_error_t kaa_tcp_channel_create(kaa_transport_channel_interface_t *self
     }
 
     /*
+     * Define type of channel (bootstrap or operations)
+     */
+    kaa_tcp_channel->channel_operation_type = KAA_SERVER_OPERATIONS;
+    if (kaa_tcp_channel->supported_service_count == 1
```

Comment:
To avoid possible misunderstandings due user inattention, I propose to walk through all supported services and if one of them is KAA_SERVICE_BOOTSTRAP, the channel will be considered as KAA_SERVER_BOOTSTRAP.